src/user.rs

```diff
@@ -1,148 +1,127 @@
 	/// Whether the user is enabled
 	pub(crate) enabled: bool,
 	/// The user's preferred username
-	pub(crate) preferred_username: StringOrBytes,
-	/// The user's LDAP ID
-	pub(crate) external_user_id: StringOrBytes,
+	pub(crate) preferred_username: Option<String>,
+	/// The user's external (non-Zitadel) ID
+	pub(crate) external_user_id: String,
 }
 
 impl User {
-	/// Convert the agnostic user to a Zitadel user
-	pub fn to_zitadel_user(&self, feature_flags: &FeatureFlags, idp_id: &str) -> ZitadelUser {
-		ZitadelUser {
-			user_data: self.clone(),
-			needs_email_verification: feature_flags.is_enabled(FeatureFlag::VerifyEmail),
-			needs_phone_verification: feature_flags.is_enabled(FeatureFlag::VerifyPhone),
-			idp_id: feature_flags.contains(&FeatureFlag::SsoLogin).then(|| idp_id.to_owned()),
-		}
-	}
-}
+	/// Convert a Zitadel user to our internal representation
+	pub fn try_from_zitadel_user(user: HumanUser, external_id: String) -> Result<Self> {
+		let first_name = user
+			.profile()
+			.and_then(|profile| profile.given_name())
+			.ok_or(anyhow!("Missing first name for {}", external_id))?
+			.clone();
 
-/// Crate-internal representation of a Zitadel user
-#[derive(Clone, Debug)]
-pub struct ZitadelUser {
-	/// Details about the user
-	pub(crate) user_data: User,
+		let last_name = user
+			.profile()
+			.and_then(|profile| profile.family_name())
+			.ok_or(anyhow!("Missing last name for {}", external_id))?
+			.clone();
 
-	/// Whether the user should be prompted to verify their email
-	pub(crate) needs_email_verification: bool,
-	/// Whether the user should be prompted to verify their phone number
-	pub(crate) needs_phone_verification: bool,
-	/// The ID of the identity provider to link with, if any
-	pub(crate) idp_id: Option<String>,
-}
+		let email = user
+			.email()
+			.and_then(|human_email| human_email.email())
+			.ok_or(anyhow!("Missing email address for {}", external_id))?
+			.clone();
 
-impl ZitadelUser {
-	/// Get a display name for the user
-	pub(crate) fn get_display_name(&self) -> String {
-		format!("{}, {}", self.user_data.last_name, self.user_data.first_name)
-	}
+		let phone = user.phone().and_then(|human_phone| human_phone.phone());
 
-	/// Return the name to be used in logs to identify this user
-	pub(crate) fn log_name(&self) -> String {
-		format!("email={}", &self.user_data.email)
+		Ok(Self {
+			first_name,
+			last_name,
+			email,
+			phone: phone.cloned(),
+			preferred_username: None,
+			external_user_id: external_id,
+			enabled: true,
+		})
 	}
 
-	/// Get idp link as required by Zitadel
-	fn get_idps(&self) -> Vec<Idp> {
-		if let Some(idp_id) = self.idp_id.clone() {
-			vec![Idp {
-				config_id: idp_id,
-				external_user_id: self.user_data.external_user_id.clone().to_string(),
-				display_name: self.get_display_name(),
-			}]
-		} else {
-			vec![]
-		}
+	/// Get a display name for this user
+	pub fn get_display_name(&self) -> String {
+		format!("{}, {}", self.last_name, self.first_name)
 	}
```

Comment:
Are we actually disregarding the `display_name` from LDAP on purpose?
For example, in tests, when we do:

```rust
	let config = ldap_config().await;
	let mut ldap = Ldap::new().await;
	ldap.create_user(
		"John",
		"Doe",
		"WHATEVER DISPLAY NAME", // Whatever we put here, it won't be picked up by the sync
		email,
		Some(phone_number),
		uid,
		enabled,
	)
	.await;

	perform_sync(config).await.expect("LDAP sync failed");
```

The display name will be ignored.

Reply:
Yes, this is as per the design: https://www.notion.so/famedly/Famedly-Sync-Agent-a7e1167785d44393a29083e81bc43bf4?pvs=4#c33cbdabda1144659da80ce2e41596b1